hbase-assembly/src/main/assembly/hadoop-three-compat.xml

```diff
@@ -0,0 +1,258 @@
+        <include>org.apache.hbase:hbase-client</include>
+        <include>org.apache.hbase:hbase-common</include>
+        <include>org.apache.hbase:hbase-endpoint</include>
+        <include>org.apache.hbase:hbase-examples</include>
+        <include>org.apache.hbase:hbase-external-blockcache</include>
+        <include>org.apache.hbase:hbase-hadoop-compat</include>
+        <include>org.apache.hbase:hbase-hadoop2-compat</include>
+        <include>org.apache.hbase:hbase-http</include>
+        <include>org.apache.hbase:hbase-it</include>
+        <include>org.apache.hbase:hbase-mapreduce</include>
+        <include>org.apache.hbase:hbase-metrics</include>
+        <include>org.apache.hbase:hbase-metrics-api</include>
+        <include>org.apache.hbase:hbase-procedure</include>
+        <include>org.apache.hbase:hbase-protocol</include>
+        <include>org.apache.hbase:hbase-protocol-shaded</include>
+        <include>org.apache.hbase:hbase-replication</include>
+        <include>org.apache.hbase:hbase-rest</include>
+        <include>org.apache.hbase:hbase-rsgroup</include>
+        <include>org.apache.hbase:hbase-server</include>
+        <include>org.apache.hbase:hbase-shell</include>
+        <include>org.apache.hbase:hbase-testing-util</include>
+        <include>org.apache.hbase:hbase-thrift</include>
+        <include>org.apache.hbase:hbase-zookeeper</include>
+      </includes>
+      <!-- Binaries for the dependencies also go in the hbase-jars directory -->
+      <binaries>
+        <outputDirectory>lib</outputDirectory>
+        <unpack>false</unpack>
+        <dependencySets>
+          <dependencySet>
+            <excludes>
+            <!-- Exclude J2EE libraries that get pulled in when building on JDK11 -->
+              <exclude>com.sun.activation:javax.activation</exclude>
+              <!-- The following artifacts are transitive dependencies of com.sun.xml.ws:jaxws-ri:pom
+                   They are needed to be included in lib/jdk11 to be added to classpath during
+                   Java 11 runtime hence excluding from main lib.
+              -->
+              <exclude>com.sun.xml.ws:*</exclude>
+              <exclude>jakarta.annotation:jakarta.annotation-api</exclude>
+              <exclude>org.glassfish.jaxb:*</exclude>
+              <exclude>com.sun.istack:istack-commons-runtime</exclude>
+              <exclude>org.glassfish.gmbal:gmbal</exclude>
+              <exclude>org.glassfish.external:management-api</exclude>
+              <exclude>org.glassfish.pfl:*</exclude>
+              <exclude>org.jvnet.staxex:stax-ex</exclude>
+              <exclude>com.sun.xml.stream.buffer:streambuffer</exclude>
+              <exclude>org.jvnet.mimepull:mimepull</exclude>
+              <exclude>com.sun.xml.fastinfoset:FastInfoset</exclude>
+              <exclude>org.glassfish.ha:ha-api</exclude>
+              <exclude>com.sun.xml.messaging.saaj:saaj-impl</exclude>
```

Comment:
The key difference is here, remove two jar from the exclude set.
<exclude>com.fasterxml.woodstox:woodstox-core</exclude>
 <exclude>org.codehaus.woodstox:stax2-api</exclude>